cxx/pclean/schema.hh

```diff
@@ -51,3 +57,31 @@ struct PCleanSchema {
   std::vector<PCleanClass> classes;
   PCleanQuery query;
 };
+
+// A class for quickly computing various properties of the schema.
+class PCleanSchemaHelper {
```

Comment:
I have mixed feelings.

On the one hand, it's nice to have one class instead of two.  On the other hand, it's nice to have the result of parsing be a plain old data (POD) struct.

The existing io.* code would have to be changed if we wanted to do work in the constructor, as right now the PCleanSchema is built up incrementally.  (Or we would have to change the compute_*_cache methods to work incrementally, which I don't think is worth doing.)

Reply:
I see -- let's leave this as-is and maybe refactor later.